src/Main.elm

```diff
@@ -55,15 +62,38 @@ update msg model =
       Tick _ -> 
         let
           newHeadPosition =
-            if model.yHeadPosition == 0 then 8
-            else model.yHeadPosition - 1
+            case model.directHead of 
+              Up ->
+                if model.yHeadPosition <= 0 then 8
+                else model.yHeadPosition - 1
+              Down ->
+                if model.yHeadPosition >= 8 then 0
+                else model.yHeadPosition + 1
         in
           ({model | yHeadPosition = newHeadPosition
                     }
           , Cmd.none
           )
-      any_other_msg ->
-        Debug.log (Debug.toString any_other_msg) <| (model, Cmd.none)
+      KeyDown _ -> 
```

Comment:
Read this
[KeyDown key](https://package.elm-lang.org/packages/ohanhi/keyboard/latest/Keyboard#KeyChange)
and this
[Key](https://package.elm-lang.org/packages/ohanhi/keyboard/latest/Keyboard#Key)